resolwe_bio/tools/make_report.py

```diff
@@ -0,0 +1,154 @@
+
+import matplotlib as mpl
+from matplotlib import pyplot as plt
+
+DECIMALS = 3  # Decimal precision when presenting results:
+
+# Set defult colors:
+GENIALIS_BLUE = '#00bcd4'
+GENIALIS_GREY = '#9E9E9E'
+mpl.rcParams['axes.labelcolor'] = GENIALIS_GREY
+mpl.rcParams['axes.edgecolor'] = GENIALIS_GREY
+mpl.rcParams['lines.color'] = GENIALIS_GREY
+mpl.rcParams['lines.antialiased'] = False
+mpl.rcParams['text.color'] = GENIALIS_GREY
+mpl.rcParams['xtick.color'] = GENIALIS_GREY
+mpl.rcParams['ytick.color'] = GENIALIS_GREY
+
+parser = argparse.ArgumentParser(description='Fill data into tex template file.')
+parser.add_argument('--sample', help='Sample name.')
+parser.add_argument('--cov', help='coverageBed report file (*.cov extension).')
+parser.add_argument('--covd', help='coverageBed report file (*.covd extension).')
+parser.add_argument('--metrics', help="CollectTargetedPcrMetrics report file.")
+parser.add_argument('--template', help='Report template file.')
+parser.add_argument('--logo', help='Logo.')
+
+
+def _format_latex(string):
+    """Format normal string to be LaTeX compliant."""
+    string = string.replace('_', '\_')
+    string = string.replace('%', '\%')
+    return string
+
+
+def parse_tsv_to_list(table_file, has_header=False, delimiter='\t'):
+    """Parse *.tsv file into list."""
+    table_data = []
+    header = None
+    with open(table_file, 'r') as tfile:
+        if has_header:
+            header = _format_latex(next(tfile).strip()).split(delimiter)
+        for line in tfile:
+            line_content = _format_latex(line.strip()).split(delimiter)
+            table_data.append(line_content)
+    return table_data, header
+
+
+def parse_targetPCRmetrics(metrics_report):
+    """Parse CollectTargetedPcrMetrics report file."""
+    with open(metrics_report) as pcr_metrics:
+        labels = []
+
+        for line in pcr_metrics:
+            if line.startswith('#'):
+                continue
+            if len(labels) == 0:
+                if line.startswith('CUSTOM_AMPLICON_SET'):
+                    labels = line.strip().split('\t')
+            else:
+                values = line.strip().split('\t')
+                break
+
+        return dict(zip(labels, values))
+
+
+def coverage_uniformity(covd, mean_coverage, cutoff):
+    """Parse coverageBed (.covd) report file."""
+    with open(covd) as pbc:
+        covered_bases = 0
+        all_bases = 0
+        for line in pbc:
+            cov_data = line.strip().split('\t')
+            if int(cov_data[6]) >= mean_coverage * cutoff:
+                covered_bases = covered_bases + 1
+            all_bases = all_bases + 1
+    return float(covered_bases) / float(all_bases)
+
+
+def make_barplot(x, y, savefile=None, dpi=300):
+    """Make barplot."""
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+
+    ax.bar(x, y, 0.5, color=GENIALIS_BLUE, linewidth=0.0)
+    ax.set_xlim(min(x), max(x))
+    ax.set_ylim(0, 1.1)
+    ax.set_xlabel(r'Amplicon')
```

Comment:
Why is a raw string needed here?

Similar as above, use double quotes for strings meant for human consumption.